### PR TITLE
19076-create-user-app: Created User model, added tests

### PR DESCRIPTION
**Pull request by firtaltv**

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[tool:pytest]
```
Comment (YevheniiaSmyrnova) [inline pytest.ini]: Why do you use it? Why you can not use standard tests?

File: static/admin/css/autocomplete.css [hunk 2]
```diff
@@ -0,0 +1,275 @@
+select.admin-autocomplete {
```
Comment (YevheniiaSmyrnova) [inline static/admin/css/autocomplete.css]: static and media folders should be in .gitignore

File: users/admin.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from django.contrib.auth.admin import UserAdmin
+from django.contrib import admin
+from .models import User
+
+
+admin.site.register(User, UserAdmin)
```
Comment (YevheniiaSmyrnova) [inline users/admin.py]: You need to add new custom fields to UserAdmin class

File: users/forms.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+from django import forms
+
+
+class ProfilePhotoForm(forms.Form):
```
Comment (YevheniiaSmyrnova) [inline users/forms.py]: Why do you need this class? You do not use it.

File: users/migrations/0003_alter_user_profile_photo.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 4.0.3 on 2022-05-03 17:04
```
Comment (YevheniiaSmyrnova) [inline users/migrations/0003_alter_user_profile_photo.py]: Please, merge 3 migration in one

File: users/tasks.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+from invoke import run, task
+
+
+@task
+def init_db(ctx):
+    pass
+
+
+@task
+def run(ctx):
```
Comment (YevheniiaSmyrnova) [inline users/tasks.py]: These tasks is empty. Why do you need an empty function?

File: users/tests/tests.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            "profile_photo": None
+        }
+
+    def test_get_user_response_code(self):
+        url = reverse('users:user_profile')
+        self.client.force_login(user=self.admin_user)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_username(self):
```
Comment (YevheniiaSmyrnova) [inline users/tests/tests.py]: This test is not needed. You do not need to add tests for each field. In one test you need to check one case.
Getting the current user - is one case. When you get the current user you need to check the response status and data.
Also, you need another test - edit the current user.

File: users/tests/user_fixtures.json [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+            "first_name": "",
+            "last_name": "",
+            "email": "admin@email.com",
+            "is_staff": "True",
+            "is_active": "True",
+            "date_joined": "2022-05-01 09:11:43.922664 +00:00",
+            "profile_photo": null
+        }
+    }
+]
```
Comment (YevheniiaSmyrnova) [inline users/tests/user_fixtures.json]: empty line at the end

File: users/views.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+from rest_framework.authentication import SessionAuthentication, BasicAuthentication
+
+from .models import User
+from .serializers import UserSerializer, LoginSerializer
+
+
+class UserProfileAPIView(RetrieveUpdateAPIView):
+    http_method_names = [
+        "get",
+        "post",
```
Comment (YevheniiaSmyrnova) [inline users/views.py]: Why do you need a post method here?

File: users/views.py [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+        "patch",
+    ]
+    authentication_classes = (SessionAuthentication, BasicAuthentication,)
```
Comment (YevheniiaSmyrnova) [inline users/views.py]: Why do you need these classes?

**Comment by YevheniiaSmyrnova** [comment]
Why do you need an empty file user/mixin.py?

File: users/tests/tests.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+        self.client.force_login(user=self.admin_user)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        content = json.loads(response.content)
+        self.assertEqual(self.test_user['username'], content['username'])
+
+    def test_edit_user(self):
+        url = reverse('users:user_profile')
+        self.client.force_login(user=self.admin_user)
+        response = self.client.patch(url, {'first_name': 'Philip'})
```
Comment (YevheniiaSmyrnova) [inline users/tests/tests.py]: Try to change the email, and make sure that the email was not changed

File: users/tests/tests.py [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+from django.urls import reverse
+from users.models import User
+
+
+class UserTest(APITestCase):
+    fixtures = ['users/tests/user_fixtures.json', ]
+
+    def setUp(self) -> None:
+        self.admin_user = User.objects.get(id=1)
+        self.test_user = {
```
Comment (YevheniiaSmyrnova) [inline users/tests/tests.py]: You do not need this, because, you already have self.admin_user

File: users/tests/tests.py [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+            "profile_photo": None
+        }
+
+    def test_get_user(self):
+        url = reverse('users:user_profile')
+        self.client.force_login(user=self.admin_user)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        content = json.loads(response.content)
+        self.assertEqual(self.test_user['username'], content['username'])
```
Comment (YevheniiaSmyrnova) [inline users/tests/tests.py]: You need to check all fields.

File: users/tests/tests.py [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+            "date_joined": "2022-05-01 09:11:43.922664 +00:00",
+            "profile_photo": None
+        }
+
+    def test_get_user(self):
+        url = reverse('users:user_profile')
+        self.client.force_login(user=self.admin_user)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        content = json.loads(response.content)
```
Comment (YevheniiaSmyrnova) [inline users/tests/tests.py]: response.data - is already dict with data

File: users/tests/tests.py [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+        self.assertEqual(self.test_user['username'], content['username'])
+
+    def test_edit_user(self):
+        url = reverse('users:user_profile')
+        self.client.force_login(user=self.admin_user)
+        response = self.client.patch(url, {'first_name': 'Philip'})
+        content = json.loads(response.content)
+        self.assertEqual('Philip', content['first_name'])
```
Comment (YevheniiaSmyrnova) [inline users/tests/tests.py]: You need to check all response data, not one field

File: users/urls.py [hunk 16]
```diff
@@ -0,0 +1,9 @@
+from django.urls import path
+from .views import UserProfileAPIView, LoginAPIView
+
+app_name = "users"
+
+urlpatterns = [
+    path('', UserProfileAPIView.as_view(), name='user_profile'),
```
Comment (YevheniiaSmyrnova) [inline users/urls.py:7]: It will be good if you use or ', or " in code.

File: users/views.py [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+    def get_object(self):
+        queryset = self.filter_queryset(self.get_queryset())
+        obj = get_object_or_404(queryset, pk=self.request.user.pk)
+        self.check_object_permissions(self.request, obj)
+        return obj
+
+
+class LoginAPIView(TokenViewBase):
+    serializer_class = LoginSerializer
+    permission_classes = (AllowAny, )
```
Comment (YevheniiaSmyrnova) [inline users/views.py]: If you did not set this, it will be allowed for any

**Comment by YevheniiaSmyrnova** [comment]
Why do you need an empty file user/forms.py?

File: pytest.ini (hunk 1 above)
Comment (firtaltv) [reply pytest.ini]: done

File: static/admin/css/autocomplete.css (hunk 2 above)
Comment (firtaltv) [reply static/admin/css/autocomplete.css]: done

File: users/admin.py (hunk 3 above)
Comment (firtaltv) [reply users/admin.py]: done

File: users/forms.py (hunk 4 above)
Comment (firtaltv) [reply users/forms.py]: done

File: users/tasks.py (hunk 6 above)
Comment (firtaltv) [reply users/tasks.py:10]: done

File: users/tests/tests.py (hunk 7 above)
Comment (firtaltv) [reply users/tests/tests.py]: done

File: users/tasks.py (hunk 6 above)
Comment (firtaltv) [reply users/tasks.py]: saved for later

File: users/tests/user_fixtures.json (hunk 8 above)
Comment (firtaltv) [reply users/tests/user_fixtures.json]: done

File: users/views.py (hunk 9 above)
Comment (firtaltv) [reply users/views.py]: done

File: users/views.py (hunk 10 above)
Comment (firtaltv) [reply users/views.py]: done

File: users/tests/tests.py (hunk 11 above)
Comment (firtaltv) [reply users/tests/tests.py]: done

File: users/tests/tests.py (hunk 12 above)
Comment (firtaltv) [reply users/tests/tests.py]: done

File: users/tests/tests.py (hunk 14 above)
Comment (firtaltv) [reply users/tests/tests.py]: done

File: users/urls.py (hunk 16 above)
Comment (firtaltv) [reply users/urls.py:7]: okay, done

File: users/views.py (hunk 17 above)
Comment (firtaltv) [reply users/views.py]: done

File: users/tests/tests.py (hunk 15 above)
Comment (firtaltv) [reply users/tests/tests.py]: done

File: users/tests/tests.py (hunk 13 above)
Comment (firtaltv) [reply users/tests/tests.py]: done

**Review by YevheniiaSmyrnova — APPROVED** [review]

**Pull request merged**